pkg/internal/itest/unixfs/generator.go

```diff
@@ -0,0 +1,134 @@
+package unixfs
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/base64"
+	"io"
+	"math/big"
+	"strings"
+	"testing"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-unixfsnode/data/builder"
+	dagpb "github.com/ipld/go-codec-dagpb"
+	"github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/linking"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/multiformats/go-multihash"
+	"github.com/stretchr/testify/require"
+)
+
+func GenerateFile(t *testing.T, linkSys *linking.LinkSystem, randReader io.Reader, size int) (cid.Cid, []byte) {
+	// a file of `size` random bytes, packaged into unixfs DAGs, stored in the remote blockstore
+	delimited := io.LimitReader(randReader, int64(size))
+	var buf bytes.Buffer
+	buf.Grow(size)
+	delimited = io.TeeReader(delimited, &buf)
+	// "size-256144" sets the chunker, splitting bytes at 256144b boundaries
+	root, _, err := builder.BuildUnixFSFile(delimited, "size-256144", linkSys)
+	require.NoError(t, err)
+	srcData := buf.Bytes()
+	rootCid := root.(cidlink.Link).Cid
+	return rootCid, srcData
+}
+
+func fileName(randReader io.Reader) (string, error) {
+	for {
+		length, err := rand.Int(randReader, big.NewInt(63))
+		if err != nil {
+			return "", err
+		}
+		name := make([]byte, length.Int64()+1)
+		_, err = randReader.Read(name)
+		if err != nil {
+			return "", err
+		}
+
+		nameStr := strings.Replace(base64.RawStdEncoding.EncodeToString(name), "/", "", -1)
+		if len(nameStr) > 0 {
+			return nameStr, nil
+		}
+	}
+}
+
+func rndInt(randReader io.Reader, max int) int {
+	coin, err := rand.Int(randReader, big.NewInt(int64(max)))
+	if err != nil {
+		return 0 // eh, whatever
+	}
+	return int(coin.Int64())
+}
+
+func GenerateDirectory(t *testing.T, linkSys *linking.LinkSystem, randReader io.Reader, targetSize int, rootSharded bool) (cid.Cid, []DirEntry) {
```

Comment:
damn this is useful. I did a whole bunch of this when I was writing stargate code -- one day we should capture this somewhere. BTW, fascinating to me that I find the go-unixfsnode code, while hardly easier, much better at these fixtures than the go-unixfs regular code.

Reply:
also worth checking out  github.com/ipfs/go-unixfsnode/data/builder/quick.